### PR TITLE
Adds add-to-queue option in Track's context menu

### PR DESCRIPTION
**Pull request by ashutoshgngwr**
Fixed bug: #545 

What got changed here :
* Add check to prevent duplicate tracks from being added to the queue
* Add add-to-queue option in Track's context menu

### CODE REVIEW

**Comment by EragonJ** [comment]
thanks @ashutoshgngwr and sorry for being late. I'll check this PR ASAP.

**Comment by ashutoshgngwr** [comment]
@EragonJ NP! Thanks! :smile:

**Comment by ashutoshgngwr** [comment]
Gently poking @EragonJ

**Comment by EragonJ** [comment]
@ashutoshgngwr sorrrrry I thought I already checked it ... I'll check it now.

File: src/views/modules/Player.js [hunk 1]
```diff
@@ -173,7 +173,12 @@ Player.prototype.addTracks = function(tracks, noUpdate) {
     return;
   }
 
-  this.tracks = [].concat(this.tracks, tracks);
+  tracks.forEach((track) => {
+    if (!this.tracks.includes(track)) {
```
Comment (EragonJ) [inline src/views/modules/Player.js]: LOL? Really ? Can you use Array.inclues() for this purpose ? I think it's only for String right ? Right now we are holding our own special Object within the array, so you need more checks I think.

File: src/views/components/shared/track/track.js [hunk 2]
```diff
@@ -107,7 +121,7 @@ class Track extends Component {
           menu.append(menuItemToRemoveTrack);
         }
         else {
-          menu.insert(0, menuItemToAddTrack);
+          menu.insert(++position, menuItemToAddTrack);
```
Comment (EragonJ) [inline src/views/components/shared/track/track.js]: can you explain a bit why you use `position` ? XD?

**Review by EragonJ — CHANGES_REQUESTED** [review]
:D

File: src/views/components/shared/track/track.js [hunk 3]
```diff
@@ -68,6 +68,20 @@ class Track extends Component {
...
     let menu = new Menu();
     let playlists = PlaylistManager.playlists;
+    let position = -1;
+
+    if (TabManager.tabName !== 'play-queue') {
+      let menuItemToAddToQueue = new MenuItem({
+        label: 'Add to Queue',
+        click: () => {
+          Player.addTracks([track]);
+          if (!Player.playingTrack) {
```
[the remark this answers is not in the document]
Comment (ashutoshgngwr) [reply src/views/components/shared/track/track.js]: Fixed!

File: src/views/components/shared/track/track.js (hunk 2 above)
Comment (ashutoshgngwr) [reply src/views/components/shared/track/track.js]: Maintains a certain hierarchy in context menu..? :sweat_smile: 
Anyways, i tried inserting at position 0 like before and it breaks without an error.

File: src/views/modules/Player.js (hunk 1 above)
Comment (ashutoshgngwr) [reply src/views/modules/Player.js]: Fixed!

**Comment by ashutoshgngwr** [comment]
Made a bit of a mess there. Feel free to squash. :sweat_smile:

File: src/views/modules/Player.js (hunk 1 above)
Comment (ashutoshgngwr) [reply src/views/modules/Player.js]: works for identical objects but I see what you mean so fixed!

**Pull request closed without merging**